x/gamm/keeper/math.go

```diff
@@ -54,6 +54,21 @@ func calcSpotPriceWithSwapFee(
 	return spotPrice.Mul(scale)
 }
 
+func solveConstantFunctionInvariant(
+	tokenBalanceFixed,
+	tokenWeightFixed,
+	tokenBalanceUnknown,
+	tokenWeightUnknown,
+	tokenAmountFixed sdk.Dec,
+) sdk.Dec {
+	weightRatio := tokenWeightFixed.Quo(tokenWeightUnknown)
+	y := tokenBalanceFixed.Quo(tokenBalanceFixed.Add(tokenAmountFixed))
+	foo := osmomath.Pow(y, weightRatio)
+
+	multiplier := sdk.OneDec().Sub(foo)
+	return tokenBalanceUnknown.Mul(multiplier)
+}
```

Comment:
Lets comment this code explaining the invariant its solving. E.g. for uniswap I'd do:

```
For fixed X_0, Y_0, X_1, solve for Y_1, where:
X_0 Y_0 = k = X_1 Y_1
Y_1 = X_0 Y_0 / X_1
```

Then we can easily match how each line in here corresponds to the equation were doing